src/views/DataManagerLogDetails.vue

```diff
@@ -0,0 +1,314 @@
+import { JobService } from "@/services/JobService";
+import logger from '@/logger';
+
+export default defineComponent ({
+  name: "DataManagerLogDetails",
+  components: {
+    IonBackButton,
+    IonBadge,
+    IonButton,
+    IonChip,
+    IonContent,
+    IonHeader,
+    IonIcon,
+    IonItem,
+    IonLabel,
+    IonList,
+    IonPage,
+    IonSpinner,
+    IonTitle,
+    IonToolbar,
+  },
+  data() {
+    return {
+      configDetails: {},
+      currentJob: {},
+      selectedFilter: 'All',
+      dataManagerLogFilters: [
+        { id: 'ALL', label: 'All' },
+        { id: 'FAILED_LOGS', label: 'Failed logs' },
+        { id: 'FAILED_RECORDS', label: 'Failed records' }
+      ],
+      dataManagerLogList: [],
+      isLoading: true,
+    }
+  },
+  computed: {
+    ...mapGetters({
+      getDataManagerLogs: 'job/getDataManagerLogs',
+      getStatusDesc: 'util/getStatusDesc',
+    }),
+  },
+  props: ['jobId'],
+  async mounted() {
+    this.currentJob = await this.fetchJobHistory(this.jobId);
+    await this.store.dispatch('job/updateCurrentJob', this.currentJob);
+    await this.fetchDataManagerConfig(this.currentJob?.runtimeData.configId)
+    await this.store.dispatch('job/fetchDataManagerLogs', this.currentJob.jobId)
+    await this.store.dispatch('job/fetchDataResource', this.getDataManagerLogs)
+    this.filterDataManagerLogs('ALL');
+    this.isLoading = false;
+  },
+  methods : {
+    filterDataManagerLogs(id) {
+      this.selectedFilter = id
+      if (id === 'ALL') {
+        this.dataManagerLogList = this.getDataManagerLogs
+      } else if (id === 'FAILED_LOGS') {
+        this.dataManagerLogList = this.getDataManagerLogs.filter(log => log.statusId === 'SERVICE_FAILED')
+      } else if (id === 'FAILED_RECORDS') {
+        this.dataManagerLogList = this.getDataManagerLogs.filter(log => log.errorRecordContentId !== null)
+      }
+    },
+    getDateTime(time) {
+      return DateTime.fromMillis(time).toFormat("dd/MM/yyyy H:mm a")
+    },
+    getProcessedFileCount() {
+      return this.getDataManagerLogs.filter((log) => log.statusId === "SERVICE_FINISHED").length
+    },
+    getErrorFileCount() {
+      return this.getDataManagerLogs.filter((log) => log.errorRecordContentId !== null).length
+    },
+    async openDownloadLogsFilePopover(dataManagerLog, event) {
+      const popover = await popoverController.create({
+        component: DownloadLogsFilePopover,
+        showBackdrop: false,
+        event: event,
+        componentProps: { dataManagerLog }
+      });
+      return popover.present()
+    },
+    async fetchJobHistory(jobId) {
+      let resp;
+
+      try {
+        resp = await JobService.fetchJobInformation({
+          "inputFields": {
+            "jobId": jobId
+          },
+          "noConditionFind": "Y"
+        })
+        if(resp.status === 200 && resp.data.docs?.length > 0 && !hasError(resp)) {
+          return resp.data.docs[0];
+        } else {
+          return [];
```

Comment:
Make consistent use of return statement, if we are returning data some section of a method then it should always return data for all the cases. In the current method we are returning data from `if` and `else` but are not returning any data from `catch` or before method execution is completed.